### PR TITLE
Add a `version` command and a version check for projects and harnesses

### PR DESCRIPTION
**Pull request by andytson-inviqa**
ws version - prints the workspace application version
the version check - checks if the workspace version satisfies composer/semver constraint in workspace().require.workspace or harness().require.workspace

projects:
```yaml
workspace('...'):
  require:
    workspace: '~0.3'
```

Harnesses:
```yaml
harness('...'):
  require:
    workspace: '~0.3'
```

### CODE REVIEW

File: docs/types/workspace.md [hunk 1]
```diff
@@ -18,17 +20,42 @@ The following attributes are automatically made available.
...
 | `workspace.description` |                                                                     |
-| `workspace.harness`     |                                                                     |
 | `namespace`             | Defaults to DNS friendly version of workspace name.                 |
 
+The following attributes can be added to change behaviour
+
+|  Key                           |  Notes                                                       |
+|--------------------------------|--------------------------------------------------------------|
+| `workspace().harness`          | A workspace harness to install                               |
+| `workspace().overlay`          | A list of ws global services to launch on `ws install`       |
```
Comment (kierenevans) [inline docs/types/workspace.md]: Description is wrong here

**Review by kierenevans — DISMISSED** [review]

**Review by kierenevans — DISMISSED** [review]

File: docs/types/harness.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+confd('harness:/'):
+  - src: template.conf
+```
+
+### A harness requiring version upgrade
+```yaml
+harness('acme-upgrade'):
+  description: Example description here
+  workspaceVersion: '~0.3'
```
Comment (kierenevans) [inline docs/types/harness.md]:
```suggestion
  require:
    workspace: '~0.3'
```
?

File: docs/types/harness.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# Type - Harness
+
+When defining a harness, a harness block is used to declare common behaviour
+
+## Declaration
+
+```
+harness('name'):
+  description: Example description here
+  workspaceVersion: '~0.3'
```
Comment (kierenevans) [inline docs/types/harness.md]:
```suggestion
  require:
    workspace: '~0.3'
```
?

File: src/Types/Harness/DefinitionFactory.php [hunk 4]
```diff
@@ -68,6 +68,7 @@ private function parseBody(array &$values, $body)
     {
         $values['description'] = $body['description'] ?? null;
         $values['require'] = $body['require'] ?? null;
+        $values['workspaceVersion'] = $body['workspaceVersion'] ?? null;
```
Comment (andytson-inviqa) [inline src/Types/Harness/DefinitionFactory.php]:
```suggestion
```

**Review by kierenevans — DISMISSED** [review]

**Review by kierenevans — DISMISSED** [review]

**Review by kierenevans — APPROVED** [review]

**Pull request merged**